ft-email-auth-provider/src/handlers/confirm_email.rs

```diff
@@ -0,0 +1,139 @@
+use crate::handlers::create_account::CreateAccount;
+use validator::ValidateEmail;
+
+#[ft_sdk::form]
+pub fn confirm_email(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+) -> ft_sdk::form::Result {
+    if !email.validate_email() {
+        return Err(ft_sdk::single_error("email", "invalid email format").into());
+    }
+
+    let (id, data) = ft_sdk::auth::provider::user_data_by_custom_attribute(
+        &mut conn,
+        crate::PROVIDER_ID,
+        crate::EMAIL_CONF_CODE_KEY,
+        &code,
+    )?;
+
+    let sent_at = data
+        .custom
+        .as_object()
+        .expect("custom is a json object")
+        .get(crate::EMAIL_CONF_SENT_AT)
+        .expect("email_conf_sent_at should exists if the account was found")
+        .as_str()
+        .expect("value must be a datetime string")
+        .parse::<chrono::DateTime<chrono::Utc>>()
+        .expect("chrono parse must work");
+
+    if key_expired(sent_at) {
+        return Err(ft_sdk::single_error("code", "confirmation code expired").into());
```

Comment:
Instead of this, automatically send a reconfirmation link and say that in the UI.

Reply:
done in https://github.com/fastn-community/auth/pull/4/commits/dd33a7eaad7ae82ac12a4867ad7fde42e8aad361